packages/@ember/-internals/container/lib/container.ts

```diff
@@ -223,8 +222,8 @@ export default class Container {
 
     assert('fullName must be a proper full name', this.registry.isValidFullName(normalizedName));
     assert(
-      'EMBER_MODULE_UNIFICATION must be enabled to pass a namespace option to factoryFor',
-      EMBER_MODULE_UNIFICATION || !options.namespace
+      'namespace must be enabled to pass a namespace option to factoryFor',
```

Comment:
I think this is kinda wrong. The message here doesn't read quite right. I think the idea here is that we can remove the entire `options.namespace` system here...

Reply:
asserts removed